pkg/server/server.go

```diff
@@ -194,7 +170,28 @@ func createApp(
 		app.Use(cors.New())
 	}
 
-	app.Use(middleware.NewNamespaceMiddleware(namespaceRepository))
+	// create namespace notification listener.
+	namespaceEventListener, err := dao.NewNamespaceListener(ctx, db.GormDB())
+	if err != nil {
+		return nil, eris.Wrap(err, "error creating namespace notification listener")
+	}
+
+	namespaceCachedRepository, err := mlflowRepositories.NewNamespaceCachedRepository(
```

Comment:
for another ticket, it seems like `NamespaceCachedRepository` should move to the `common/dao` package